### ISSUE TITLE
update variable section does not trigger redeploying vms

### ISSUE THREAD

**Issue by fredwangwang** [cloudfoundry/bosh#2208]
**Describe the bug**
After updating the certificate definition in the [variables](https://bosh.io/docs/manifest-v2/#variables) section, we triggered a bosh deployment and noticed that the certificate didn't actually get updated and the VMs referencing the certificate didn't get redeployed, though BOSH says deployment was successful.

**To Reproduce**
Steps to reproduce the behavior (example):
1. Deploy a bosh deployment that uses a certificate variable, e.g.
  ```yml
  - name: /concourse/main/docker-tls
    type: certificate
    options:
      ca: /concourse/main/docker-ca
      common_name: docker-host-ca
      alternative_names: [ '10.0.1.1' ]
  ```
2. Update the certificate definition:
  ```diff
  -     alternative_names: [ '10.0.1.1' ]
  +     alternative_names: [ '10.0.1.1', '10.0.1.2' ]
  ```
3. Redeploy
4. BOSH correctly recognize the alternative_names, changed, and ask fro Continue
5. Confirm, and after several seconds, deployment finishes, change didn't get applied. The affected vm didn't redeploy, the cert in credhub didn't update. `openssl -in <(credhub get -n /concourse/main/docker-tls) -noout -text` shows the original alternative names.

**Expected behavior**
certificate in credhub get updated, affected vms redeployed with the new cert.

**Versions (please complete the following information):**
 - Infrastructure: gcp
 - BOSH version: 270.3.0
 - BOSH CLI version: 5.5.1
 - Stemcell version: xenial/315.64

**Deployment info:**
basically: https://github.com/cloudfoundry-incubator/docker-boshrelease/blob/master/manifests/containers/example.yml

**Comment by mrosecrance** [comment]
Have you experimented with the [update_mode](https://bosh.io/docs/manifest-v2/#variables)? By default, we set the mode to `no-overwrite` which doesn't do a regenerate if the credential is found. It sounds like you're looking for `converge` mode

**Comment by fredwangwang** [comment]
Ohhh thanks! Totally didn't notice that thing.
